### PR TITLE
Bg user story 30

### LINKED ISSUE #33: User Story 30, User cancels an order
As a registered user
When I visit an order's show page
I see a button or link to cancel the order 
When I click the cancel button for an order, the following happens:
- Each row in the "order items" table is given a status of "unfulfilled"
- The order itself is given a status of "cancelled"
- Any item quantities in the order that were previously fulfilled have their quantities returned to their respective merchant's inventory for that item.
- I am returned to my profile page
- I see a flash message telling me the order is now cancelled
- And I see that this order now has an updated status of "cancelled"

### PR DESCRIPTION
**Pull request by brian-greeson**
## User can cancel an order
Closes: #33 
### Added 
- fulfill, unfulfill && fulfilled, unfulfilled? methods to ItemOrder model
- Updated route prefixes from `profile` to `user`
- Refactored Order status logic

### CODE REVIEW

File: app/models/item_order.rb [hunk 1]
```diff
@@ -7,4 +7,29 @@ class ItemOrder <ApplicationRecord
...
+
+  def fulfill
+    item.update(inventory: item.inventory - quantity) if unfulfilled?
+    update(status: "fulfilled")
+    order.update(status: "packaged") if items.where(status: "unfulfilled").any?
+  end
+  
+  def fulfilled?
+    return true if status == "fulfilled"
+    false
```
Comment (kelshadarby) [inline app/models/item_order.rb:19]: Is this false necessary for the guard clause?

File: spec/features/user/orders/show_spec.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 RSpec.describe "As a default user", type: :feature do
   describe "When I view an orders show page" do
-    #===============================================================================
```
Comment (kelshadarby) [inline spec/features/user/orders/show_spec.rb:5]: Nice catch!

**Review by kelshadarby — APPROVED** [review]
This all looks great! Nice work!

File: app/models/order.rb [hunk 3]
```diff
@@ -13,4 +13,11 @@ def total_items
     item_orders.sum('quantity')
   end
   
+  def cancel
+    item_orders.each do |item_order|
+      item_order.unfulfill
+    end
+    self.update(status: "canceled")
```
Comment (fredrondina) [inline app/models/order.rb:20]: This is great

**Review by fredrondina — APPROVED** [review]
Fantastic work on this, looks great

**Pull request merged**